hugolib/site_test.go

```diff
@@ -296,6 +296,42 @@ func TestDraftAndFutureRender(t *testing.T) {
 	viper.Set("BuildFuture", false)
 }
 
+func TestFutureExpirationRender(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+
+	hugofs.InitMemFs()
+	sources := []source.ByteSource{
+		{filepath.FromSlash("sect/doc3.md"), []byte("---\ntitle: doc1\nexpirydate: \"2400-05-29\"\n---\n# doc1\n*some content*")},
+		{filepath.FromSlash("sect/doc4.md"), []byte("---\ntitle: doc2\nexpirydate: \"2000-05-29\"\n---\n# doc2\n*some content*")},
+	}
+
+	siteSetup := func() *Site {
+		s := &Site{
+			Source: &source.InMemorySource{ByteSource: sources},
+		}
+
+		s.initializeSiteInfo()
+
+		if err := s.createPages(); err != nil {
+			t.Fatalf("Unable to create pages: %s", err)
+		}
+		return s
+	}
+
+	viper.Set("baseurl", "http://auth/bub")
+
+	s := siteSetup()
+
+	if len(s.Pages) > 1 {
+		t.Fatal("Expired content published unexpectedly")
+	}
+
+	if len(s.Pages) < 1 {
+		t.Fatal("Valid content expired unexpectedly")
+	}
```

Comment:
These two assertions looks a little bit funky. What you want to know is that len(Pages) == 1 and s.Pages[0].title = "doc2"